pkg/gadgets/run/tracer/run.go

```diff
@@ -123,9 +123,87 @@ func getGadgetInfo(params *params.Params, args []string, logger logger.Logger) (
 		}
 	}
 
+	log.Warnf("GadgetMetadata.EBPFParams: %v", ret.GadgetMetadata.EBPFParams)
+	if err := fillTypeHints(spec, ret.GadgetMetadata.EBPFParams); err != nil {
+		return nil, fmt.Errorf("fill parameters type hints: %w", err)
+	}
+
 	return ret, nil
 }
 
+func getTypeHint(typ btf.Type) params.TypeHint {
+	switch typedMember := typ.(type) {
+	case *btf.Int:
+		switch typedMember.Encoding {
+		case btf.Signed:
+			switch typedMember.Size {
+			case 1:
+				return params.TypeInt8
+			case 2:
+				return params.TypeInt16
+			case 4:
+				return params.TypeInt32
+			case 8:
+				return params.TypeInt64
+			}
+		case btf.Unsigned:
+			switch typedMember.Size {
+			case 1:
+				return params.TypeUint8
+			case 2:
+				return params.TypeUint16
+			case 4:
+				return params.TypeUint32
+			case 8:
+				return params.TypeUint64
+			}
+		case btf.Bool:
+			return params.TypeBool
+		case btf.Char:
+			return params.TypeUint8
+		}
+	case *btf.Float:
+		switch typedMember.Size {
+		case 4:
+			return params.TypeFloat32
+		case 8:
+			return params.TypeFloat64
+		}
+	case *btf.Typedef:
+		typ, err := getUnderlyingType(typedMember)
+		if err != nil {
+			return params.TypeUnknown
+		}
+		return getTypeHint(typ)
+	case *btf.Volatile:
+		return getTypeHint(typedMember.Type)
+	}
+
+	return params.TypeUnknown
+}
+
+// fillTypeHints fills the TypeHint field in the ebpf parameters according to the BTF information
+// about those constants.
+func fillTypeHints(btfSpec *ebpf.CollectionSpec, params map[string]types.EBPFParam) error {
```

Comment:
nit: we call if `spec` (instead of btfSpec) in all other places. (btw, it's not about btf but about bpf on this case)